modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -0,0 +1,425 @@
+import org.mule.extension.introspection.Extension;
+import org.mule.extension.introspection.Operation;
+import org.mule.extension.introspection.Parameter;
+import org.mule.extension.runtime.ConfigurationInstanceProvider;
+import org.mule.extension.runtime.ConfigurationInstanceRegistrationCallback;
+import org.mule.extension.runtime.OperationContext;
+import org.mule.extension.runtime.OperationExecutor;
+import org.mule.module.extension.internal.introspection.DefaultExtensionFactory;
+import org.mule.module.extension.internal.introspection.ExtensionDiscoverer;
+import org.mule.module.extension.internal.runtime.ConfigurationObjectBuilder;
+import org.mule.module.extension.internal.runtime.DelegatingOperationExecutor;
+import org.mule.module.extension.internal.runtime.StaticConfigurationInstanceProvider;
+import org.mule.module.extension.internal.runtime.resolver.EvaluateAndTransformValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.ResolverSet;
+import org.mule.module.extension.internal.runtime.resolver.StaticValueResolver;
+import org.mule.module.extension.internal.runtime.resolver.ValueResolver;
+import org.mule.util.ObjectNameHelper;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation of {@link ExtensionManagerAdapter}
+ *
+ * @since 3.7.0
+ */
+public final class DefaultExtensionManager implements ExtensionManagerAdapter, MuleContextAware, Initialisable
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultExtensionManager.class);
+
+    private final ExtensionRegister register = new ExtensionRegister();
+    private final ServiceRegistry serviceRegistry = new SPIServiceRegistry();
+
+    private MuleContext muleContext;
+    private ObjectNameHelper objectNameHelper;
+    private ExtensionDiscoverer extensionDiscoverer = new DefaultExtensionDiscoverer(new DefaultExtensionFactory(serviceRegistry), serviceRegistry);
+
+    /**
+     * Searches the mule registry for instances of {@link ConfigurationInstanceProvider}
+     * and registers them through the {@link #registerConfigurationInstanceProvider(String, ConfigurationInstanceProvider)}
+     * method
+     */
+    @Override
+    public void initialise() throws InitialisationException
+    {
+        for (Map.Entry<String, ConfigurationInstanceProvider> instanceProviderEntry : muleContext.getRegistry().lookupByType(ConfigurationInstanceProvider.class).entrySet())
+        {
+            ConfigurationInstanceProvider<?> instanceProvider = instanceProviderEntry.getValue();
+            registerConfigurationInstanceProvider(instanceProviderEntry.getKey(), instanceProvider);
+        }
+    }
+
+    @Override
+    public List<Extension> discoverExtensions(ClassLoader classLoader)
+    {
+        LOGGER.info("Starting discovery of extensions");
```

Comment:
I think yes because this would not happen very often and when it does you sure wanna know about it